cdap-metadata-ext-spanner/src/main/java/io/cdap/cdap/metadata/spanner/TableUpdates.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.metadata.spanner;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.data.schema.SchemaWalker;
+import io.cdap.cdap.api.metadata.MetadataEntity;
+import io.cdap.cdap.api.metadata.MetadataScope;
+import io.cdap.cdap.spi.metadata.Metadata;
+import io.cdap.cdap.spi.metadata.MetadataConstants;
+import io.cdap.cdap.spi.metadata.ScopedName;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A helper class to process and prepare metadata fields for optimized storage.
+ * This class encapsulates the logic for flattening text fields, parsing schemas,
+ * and extracting specific properties for relational storage.
+ */
+public class TableUpdates {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TableUpdates.class);
+
+  private final MetadataEntity entity;
+
+  // Fields directly mapped to metadata table columns
+  private final String namespace;
+  private final String type;
+  private final String name;
+  private final Long created;
+  private final StringBuilder userText;
+  private final StringBuilder systemText;
+  private final Set<Property> metadataProps;
+
+  /**
+   * Constructs and processes all metadata for a given entity.
+   *
+   * @param entity the metadata entity being processed.
+   * @param metadata the metadata to be associated with the entity.
+   * @throws IOException if schema parsing fails.
+   */
+  public TableUpdates(MetadataEntity entity, Metadata metadata) throws IOException {
+    this.entity = entity;
+    this.namespace = entity.getValue("namespace");
+    this.type = entity.getType().toLowerCase();
+    this.name = Objects.requireNonNull(entity.getValue(entity.getType())).toLowerCase();
+
+    Map<ScopedName, String> processedProperties = new HashMap<>();
+    Set<Property> processedPropsWithSummary = processAndSummarizeProperties(metadata.getProperties(),
```

Comment:
Instead of updating map in different function, use something like : 

```
Map<ScopedName, String> processedProperties = Map<String, String> result = metadata.getProperties().entrySet().stream()
    .filter(entry -> MetadataConstants.SCHEMA_KEY.equals(entry.getKey()))
    .collect(Collectors.toMap(
        Map.Entry::getKey,
        entry -> reformatSchemaProperty(entry.getValue())
    ));
    ```